backend/server.js

```diff
@@ -1,29 +1,107 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import crypto from "crypto";
+import bcrypt from "bcrypt";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+mongoose.set("useCreateIndex", true);
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+const User = mongoose.model("User", {
+  name: {
+    type: String,
+    unique: true,
+    required: true,
+    minlength: 5,
```

Comment:
Maybe the minLength for username is unnecessary? I (Sara) was not able to make an account using only my first name.
Since the name is not hidden or encrypted it should matter if it's long or short, right? 🤔

Reply:
It is not a good practice to let people use short inputs when you are creating a unique field. So `minlength` is a must in this situation 😸